gpjax/gps.py

```diff
@@ -664,7 +669,7 @@ def predict(
         Lx_inv_Kxt = Lx.solve(Ktx.T)
 
         # Whitened function values, wx, corresponding to the inputs, x
-        wx = self.latent
+        wx = self.get_latent(num_datapoints=n)
```

Comment:
Surely self._latent is already built (i.e not None) at this point.  passing n doesnt do anything?  Same below as well

Reply:
perhaps n should be optional ?

Reply:
There is no longer a guarantee that `self._latent` will be built at this point as there is no longer a `__post_init__` method (which relied on the `num_datapoints` field of the likelihood). I think that making `n` not optional could also be beneficial for ensuring that the number of datapoints used in the dataset and latent function are also consistent?

Reply:
Agreed with @Thomas-Christie here that under this approach, we'd need such an argument. The real crux of this is the earlier thread with Dan and Thomas.